azurerm/internal/services/network/resource_arm_application_gateway.go

```diff
@@ -1608,6 +1622,10 @@ func resourceArmApplicationGatewayRead(d *schema.ResourceData, meta interface{})
 		if setErr := d.Set("waf_configuration", flattenApplicationGatewayWafConfig(props.WebApplicationFirewallConfiguration)); setErr != nil {
 			return fmt.Errorf("Error setting `waf_configuration`: %+v", setErr)
 		}
+
+		if setErr := d.Set("firewall_policy_id", flattenApplicationGatewayFirewallPolicy(props.FirewallPolicy)); setErr != nil {
+			return fmt.Errorf("Error setting `firewall_policy_id`: %+v", setErr)
+		}
```

Comment:
given this is no longer a block or complex set we can:
```suggestion
		if props.FirewallPolicy !=nil {
				d.Set("firewall_policy_id", props.FirewallPolicy.ID)		
		}
```